src/swarm-util/Swarm/Util.hs

```diff
@@ -325,6 +326,18 @@ unsnocNE (x :| xs) = go x xs
   go y [] = ([], y)
   go y (z : zs) = let ~(ws, w) = go z zs in (y : ws, w)
 
+-- | Iterate a function a specific number of times.
+--
+-- >>> iterateN 4 succ 'a'
+-- 'e'
+-- >>> iterateN 0 succ 'a'
+-- 'a'
+-- >>> iterateN 5 (*2) 1
+-- 32
+iterateN :: Int -> (a -> a) -> a -> a
+iterateN 0 _ = id
+iterateN n f = iterateN (n - 1) f . f
```

Comment:
This could cause confusion - containers use this name to construct a Seq.

How about applyN or nTimes?
https://hoogle.haskell.org/?hoogle=Int%20-%3E%20(a%20-%3E%20a)%20-%3E%20a%20-%3E%20a